src/main/java/Duke.java

```diff
@@ -1,10 +1,111 @@
+import task.*;
+
+import java.io.*;
+import java.text.ParseException;
+
 public class Duke {
+
+    private static final String FILEPATH = "duke.txt";
+    private Ui ui;
+    private Storage storage;
+    private TaskList taskList;
+
+    /**
+     * Constructor for Duke class.
+     * @param filePath Path for the Duke save file.
+     */
+    public Duke(String filePath) {
+        //storage = new Storage(filePath);
+        taskList = Storage.loadTasks(FILEPATH);
+        ui = new Ui();
+    }
+
+    /**
+     * Method to run the Duke program.
+     */
+    public void run() {
+        Ui.printWelcome();
+        while(true){
```

Comment:
Perhaps you could add whitespace immediately after the brackets here.I noticed this issue in other areas as well but I believe this was just an oversight on your part.